homeassistant/components/homekit_controller/humidifier.py

```diff
@@ -0,0 +1,224 @@
+"""Support for HomeKit Controller humidifier."""
+from typing import Any, Dict, List, Optional
+
+from aiohomekit.model.characteristics import CharacteristicsTypes
+
+from homeassistant.components.humidifier import HumidifierEntity
+from homeassistant.components.humidifier.const import (
+    ATTR_AVAILABLE_MODES,
+    ATTR_HUMIDITY,
+    ATTR_MAX_HUMIDITY,
+    ATTR_MIN_HUMIDITY,
+    ATTR_MODE,
+    DEVICE_CLASS_DEHUMIDIFIER,
+    DEVICE_CLASS_HUMIDIFIER,
+    MODE_AUTO,
+    SUPPORT_MODES,
+)
+from homeassistant.core import callback
+
+from . import KNOWN_DEVICES, HomeKitEntity
+
+SUPPORT_FLAGS = 0
+
+ATTR_DEHUMIDIFIER_THRESHOLD = "dehumidifier_threshold"
+ATTR_HUMIDIFIER_THRESHOLD = "humidifier_threshold"
+ATTR_CURRENT_HUMIDITY = "current_humidity"
+
+MODE_OFF = "off"
+MODE_HUMIDIFYING = "humidifying"
+MODE_DEHUMIDIFYING = "dehumidifying"
+
+HK_MODE_TO_HA = {
+    0: MODE_OFF,
+    1: MODE_AUTO,
+    2: MODE_HUMIDIFYING,
+    3: MODE_DEHUMIDIFYING,
+}
+
+HA_MODE_TO_HK = {
+    MODE_AUTO: 0,
+    MODE_HUMIDIFYING: 1,
+    MODE_DEHUMIDIFYING: 2,
+}
+
+
+class HomeKitHumidifier(HomeKitEntity, HumidifierEntity):
+    """Representation of a HomeKit Controller Humidifier."""
+
+    def __init__(self, accessory, devinfo):
+        """Initialize a HomeKit humidifier entity."""
+        super().__init__(accessory, devinfo)
+
+        dehumidifier_threshold = self.service.value(
```

Comment:
You can use `self.service.has(CharacteristicsTypes.RELATIVE_HUMIDITY_DEHUMIDIFIER_THRESHOLD)` here I think?

Reply:
In general i've been trying to avoid setting aything in `__init__` and instead checking against the aiohomekit model directly. This is so that if a firmware update adds something HA picks it up without a restart. I don't think a humidifier is likely to get a firmware update to become a dehumidifier, but i'd like to stick to the pattern for consistency unless you have a good reason.

You could add a `@property` as a shortcut if you want to avoid some repetition?